byterun/caml/roots.h

```diff
@@ -39,6 +39,12 @@ CAMLextern void caml_do_local_roots(scanning_action f, char * bottom_of_stack,
 
 CAMLextern void (*caml_scan_roots_hook) (scanning_action);
 
+#ifdef NATIVE_CODE
+/* executed just before calling the entry point of a dynamically
+   loaded native code module. */
+CAMLextern void (*natdynlink_hook)(void* handle, char* unit);
+#endif
+
```

Comment:
Why is it in `roots.h`? What does it have to do with roots? Why not `osdeps.h` or `misc.h`? There's no `natdynlink.h` so you should probably create a `hooks.h` and start gathering all the hooks there.